### PR TITLE
feat: use batch RPC queries

### PR DESCRIPTION
**Pull request by samsiegart**
refs https://github.com/Agoric/agoric-sdk/issues/6870

### CODE REVIEW

File: patches/@agoric+smart-wallet+0.5.3.patch [hunk 1]
```diff
@@ -0,0 +1,14 @@
+diff --git a/node_modules/@agoric/smart-wallet/src/marshal-contexts.js b/node_modules/@agoric/smart-wallet/src/marshal-contexts.js
+index de5b3b9..77173f1 100644
+--- a/node_modules/@agoric/smart-wallet/src/marshal-contexts.js
++++ b/node_modules/@agoric/smart-wallet/src/marshal-contexts.js
+@@ -286,7 +286,8 @@ export const makeImportContext = (makePresence = defaultMakePresence) => {
+      * @param {string} iface
+      */
+     fromBoard: (slot, iface) => {
+-      isDefaultBoardId(slot) || Fail`bad board slot ${q(slot)}`;
++      // Don't throw on null slots.
```
Comment (turadg) [inline patches/@agoric+smart-wallet+0.5.3.patch:10]: We shouldn't have to patch our own deps.

We could `makeImportContext` taking an option for whether to throw on null slots, but the context shouldn't be concerned with whether the slot is null. When the slot is `null`, why call `fromBoard` at all?

Please remove this patch and make the `const context = makeImportContext().fromBoard` into something that knows how to handle null slots.

File: src/vite-env.d.ts [hunk 2]
```diff
@@ -15,13 +15,6 @@ declare module '@agoric/notifier' {
   export const makeAsyncIterableFromNotifier;
 }
 
-declare module '@agoric/casting' {
```
Comment (turadg) [inline src/vite-env.d.ts:18]: yay

**Review by turadg — CHANGES_REQUESTED** [review]

File: patches/@agoric+smart-wallet+0.5.3.patch (hunk 1 above)
Comment (samsiegart) [reply patches/@agoric+smart-wallet+0.5.3.patch:10]: We do this patch in dapp-inter and wallet-app as well. Can I make a separate issue to address that and not block this PR on it?

Comment (turadg) [reply patches/@agoric+smart-wallet+0.5.3.patch:10]: Okay

**Review by turadg — APPROVED** [review]

**Pull request merged**